lib/etcdv3/lease_keep_alive.rb

```diff
@@ -0,0 +1,184 @@
+class Etcdv3
+    # A session is used to keep a lease alive for the duration
+  # of a connection with an etcd3 server.
+  #
+  # This is useful for implementing functionality, such
+  # as distributed locks, where it may be necessary to
+  # maintain a lock as long as the connection is alive
+  # and well.
+  #
+  # Session provides a way to hook into events using a
+  # listener. This will allow our applications to respond
+  # in case we let a lease expire accidentally or there
+  # is a connection issue and we need to close the session.
+  #
+  # Parameters
+  # stub - the client stub to send requests to
+  # keep_alive_padding - how much padding to give to the refresh calls
```

Comment:
Out of curiosity, what's the benefit of making `keep_alive_padding` tunable?

Reply:
Not sure if there is much value here or not, but wanted to keep it configurable in case there were. One case I can see where it would be useful is if the TTL is short, like 5 seconds, and you wanted to have just a 2 seconds padding or something.